build.sbt

```diff
@@ -18,6 +18,7 @@ libraryDependencies += "com.databricks" % "dbutils-api_2.12" % "0.0.5" % Provide
 libraryDependencies += "com.amazonaws" % "aws-java-sdk-s3" % "1.11.595" % Provided
 libraryDependencies += "io.delta" % "delta-core_2.12" % "1.0.0" % Provided
 libraryDependencies += "org.scalaj" %% "scalaj-http" % "2.4.2"
+libraryDependencies += "com.lihaoyi" %% "sourcecode" % "0.4.1"
 
```

Comment:
I guess this PR also contains the framework changes which has "sourcecode" jar. Can we sagrigate these 2 different things and only have the changes which will improve the performance.

Reply:
after we spoke this morning I realized how much work it will be to revert the changes associated with #1223 here so I suggest that we merge it into `0820_release` first to simplify things, please.  then the deltas will be clearer when comparing the branches as part of this PR.